### PR TITLE
Alpide raw data encoding/decoding + raw data clusterization

### PR DESCRIPTION
**Pull request by shahor02**

### CODE REVIEW

**Comment by Barthelemy** [comment]
Hi Ruben, 
With this PR, would it be possible to take simulated ITS data and encode them in a format similar to what we will get on the GBT ?

**Comment by shahor02** [comment]
Hi Barth

Yes, but at the moment the format above the single chip level is improvised.

Cheers
 Ruben

File: Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx [hunk 1]
```diff
@@ -172,106 +170,86 @@ void Clusterer::updateChip(UInt_t ip)
...
-        const auto& dig = mPixels[next];
-        const auto pix = pixData[dig.second]; // PixelData
-        x += pix.getRowDirect();
-        z += pix.getCol();
+        mPixArrBuff[npix++] = pix; // needed for cluster topology
         if (pix.getRowDirect() < rowMin) {
           rowMin = pix.getRowDirect();
-        }
-        if (pix.getRowDirect() > rowMax) {
+        } else if (pix.getRowDirect() > rowMax) {
```
Comment (iouribelikov) [inline Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx]: I believe, this change should be undone. The checks with respect to rowMin and rowMax must be fully independent.

File: Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx [hunk 2]
```diff
@@ -172,106 +170,86 @@ void Clusterer::updateChip(UInt_t ip)
...
           rowMax = pix.getRowDirect();
         }
         if (pix.getCol() < colMin) {
           colMin = pix.getCol();
-        }
-        if (pix.getCol() > colMax) {
+        } else if (pix.getCol() > colMax) {
```
Comment (iouribelikov) [inline Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx]: I believe, this change should be undone. The checks with respect to colMin and colMax must be fully independent.

File: Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx [hunk 3]
```diff
@@ -172,106 +170,86 @@ void Clusterer::updateChip(UInt_t ip)
...
+      }
+      next = mPreClusterHeads[i2];
+      while (next >= 0) {
+        const auto& dig = mPixels[next];
+        const auto pix = pixData[dig.second]; // PixelData
+        if (npix < mPixArrBuff.size()) {
+          mPixArrBuff[npix++] = pix; // needed for cluster topology
+          if (pix.getRowDirect() < rowMin) {
+            rowMin = pix.getRowDirect();
+          } else if (pix.getRowDirect() > rowMax) {
```
Comment (iouribelikov) [inline Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx]: The checks with respect to rowMin and rowMax must be independent.  No "else", right ?

File: Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx [hunk 4]
```diff
@@ -172,106 +170,86 @@ void Clusterer::updateChip(UInt_t ip)
...
+            rowMax = pix.getRowDirect();
+          }
+          if (pix.getCol() < colMin) {
+            colMin = pix.getCol();
+          } else if (pix.getCol() > colMax) {
```
Comment (iouribelikov) [inline Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx]: The checks with respect to colMin and colMax must be independent. No "else", right ?

File: Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/Clusterer.h [hunk 5]
```diff
@@ -90,8 +92,59 @@ class Clusterer
...
+    if (pix.getRowDirect() < rMin) {
+      rMin = pix.getRowDirect();
+    } else if (pix.getRowDirect() > rMax) {
+      rMax = pix.getRowDirect();
+    }
+    if (pix.getCol() < cMin) {
+      cMin = pix.getCol();
+    } else if (pix.getCol() > cMax) {
+      cMax = pix.getCol();
+    }
```
Comment (iouribelikov) [inline Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/Clusterer.h]: The comparisons with rMin, rMax, cMin and cMax must be independent.  No "else" between the if's.

**Pull request merged**